### PR TITLE
GN-5005: rework snippet data-model and add support for reordering snippets in a snippet-list

### PR DESCRIPTION
**Pull request by elpoelma**
## Overview
This PR adjusts and simplifies the data-model of snippets and snippet-lists.
Specifically the following changes are included in this PR:

**Removal of distinction between published-snippet-container/published-snippet and their related document-container/editor-document models**
The `editor-document` and `document-container` models are no longer used for snippets.
The `published-snippet-container` and `published-snippet` models are removed. 
They are replaced by 3 new simple models:
- `snippet`: replaces both `published-snippet-container` and `document-container`
- `snippet-version`: replaces both `published-snippet` and `editor-document`
- `snippet-list`: contains a list of `snippet` resources

**Addition of a `position` property to the `snippet` model and support for manually reordering snippets**
- Optional, `created-on` is used as a fallback for sorting
- Allows for creators to manually reorder snippets in a snippet-list (using drag-and-drop)
   * Uses the `ember-sortable` package
   * Added a very light css animation to the drag-and-drop as I was feeling fancy :sunglasses: 

TODOS:
- [x] Update to stable version of editor-plugins

##### connected issues and PRs:
[GN-5005](https://binnenland.atlassian.net/browse/GN-5005?atlOrigin=eyJpIjoiNmJjODI4OTg4ZjgyNGU0Y2FmYmY1ZmY3OGVmZjliNzYiLCJwIjoiaiJ9)
https://github.com/lblod/app-reglementaire-bijlage/pull/84
https://github.com/lblod/ember-rdfa-editor-lblod-plugins/pull/482
https://github.com/lblod/frontend-gelinkt-notuleren/pull/720

### Setup
**Set-up the RB stack**

<details>
    <summary>docker-compose.override.yml</summary>

   ```yaml
    frontend:
      image: !reset null
      build: https://github.com/lblod/frontend-reglementaire-bijlage.git#GN-5005-ordering-of-fragments
   ```
</details>

**Set-up the GN stack**
<details>
    <summary>docker-compose.override.yml</summary>
    
   ```yaml
    editor:
      image: !reset null
      build: https://github.com/lblod/frontend-gelinkt-notuleren.git#GN-5005-snippet-rework
      environment:
        EMBER_REGULATORY_STATEMENT_ENDPOINT: "http://rb.localhost/raw-sparql"
        EMBER_REGULATORY_STATEMENT_FILE_ENDPOINT: "http://rb.localhost/files"
   ```
</details>

### How to test/reproduce
- Set-up both the RB stacks and GN stacks and connect them to each other

#### Starting from an empty database
- Create snippet-lists and snippets should works as expected
- You should be able to manually reorder snippets in a snippet-list
- In the template editor, you should be able to configure snippet placeholders with snippet lists
- Imported resources should work as expected
- You should be able to fill-in the snippet placeholders (with imported resources) correctly on the GN side
- When filling in snippet placeholders, notice that the snippets are consistently ordered

#### Starting from a QA/DEV backup
- Ensure the migration moves the snippets correctly to the new model
- If a snippet has no position, they are ordered by creation date (in descending order)
- Reordering a single snippet in an older list should assign positions to all other snippets

### Challenges/uncertainties
- I did not use [ember-drag-drop](https://github.com/adopted-ember-addons/ember-drag-drop), but [ember-sortable](https://github.com/adopted-ember-addons/ember-sortable) to implement the drag-and-drop of the snippets. It seems to be more maintained, more modern and easier to use.
- Snippets are never really deleted, but are marked as invalid. Snippet lists are fully deleted though.

### Checks PR readiness
- [x] UI: works on smaller screen sizes
- [x] UI: feedback for any loading/error states
- [x] Check cancel/go-back flows
- [x] Check database state correct when deleting/updating (especially regarding relationships)
- [ ] changelog
- [x] npm lint

### CODE REVIEW

**Review by piemonkey — COMMENTED** [review]
looks good. Mostly very minor things, except that one weird error.

File: app/components/snippet-list-form.hbs [hunk 1]
```diff
@@ -39,47 +36,71 @@
...
+    </thead>
+    <tbody {{sortable-group onChange=this.reorderSnippets}}>
+      {{#if this.snippetsRequest.isLoading}}
+        <tr><td colspan="100%">{{t "utility.loading"}}</td></tr>
+      {{else if this.snippetsRequest.isError}}
+        <tr><td colspan="100%">{{t 'snippets.edit-snippet-list.table.loading-error'}}
+            <AuButton
+              @skin="link"
+              {{on "click" this.snippetsRequest.retry}}
+            >{{t 'utility.retry'}}</AuButton></td></tr>
```
[the remark this answers is not in the document]
Comment (elpoelma) [reply app/components/snippet-list-form.hbs:57]: Agreed :smile: It's also not difficult to do as we kind of get it for free when using tasks/trackedFunctions

File: app/components/app-chrome.js [hunk 2]
```diff
@@ -22,6 +24,16 @@ export default class AppChromeComponent extends Component {
     return status;
   }
 
+  get updatedOn() {
+    if (this.editorDocument instanceof EditorDocumentModel) {
```
[the remark this answers is not in the document]
Comment (elpoelma) [reply app/components/app-chrome.js]: I refactored the `app-chrome` component a bit:
- Renamed `@editorDocument` to `@document`
- Added basic jsdoc types for the arguments
- Removed unused code

File: app/components/snippet-list-form.hbs [hunk 3]
```diff
@@ -39,47 +36,71 @@
...
-          </AuButton>
-        </td>
-      </c.body>
-    </s.content>
-  </AuDataTable>
+      </tr>
+    </thead>
+    <tbody {{sortable-group onChange=this.reorderSnippets}}>
+      {{#if this.snippetsRequest.isLoading}}
+        <tr><td colspan="100%">{{t "utility.loading"}}</td></tr>
```
[the remark this answers is not in the document]
Comment (elpoelma) [reply app/components/snippet-list-form.hbs:51]: We could move to `AuLoader` indeed, but we should then do so for all tables. At the moment, I prefer to keep consistency across the different tables in the application.

File: app/components/snippet-list-form.hbs [hunk 4]
```diff
@@ -39,47 +36,71 @@
...
+      {{else if this.snippetsRequest.isError}}
+        <tr><td colspan="100%">{{t 'snippets.edit-snippet-list.table.loading-error'}}
```
[the remark this answers is not in the document]
Comment (elpoelma) [reply app/components/snippet-list-form.hbs]: A neutral ('info') au-alert seems to fit well.

File: app/controllers/snippet-management/edit/edit-snippet.js [hunk 5]
```diff
@@ -395,36 +391,4 @@ export default class SnippetManagementEditSnippetController extends Controller {
     );
     return saveCollatedImportedResources(list);
```
[the remark this answers is not in the document]
Comment (elpoelma) [reply app/controllers/snippet-management/edit/edit-snippet.js:393]: The cause of this issue lies with the sparse fieldsets used to fetch snippets on the snippet overview page.
The attributes of the `snippet-version` models were not being correctly loaded.

As we need all attributes for deletion and saving, I'm wondering if using sparse-fieldsets is worth it, as we still need to fetch all contents if we want to delete/save a snippet.

**Pull request merged**